### PR TITLE
samples: bluetooth: Make ECC thread params configurable

### PR DESCRIPTION
**Pull request by pdunaj**
Make ECC thread stack size and priority configurable.

Signed-off-by: Pawel Dunaj <pawel.dunaj@nordicsemi.no>

### CODE REVIEW

File: subsys/bluetooth/host/hci_ecc.c [hunk 1]
```diff
@@ -321,6 +321,7 @@ void bt_hci_ecc_init(void)
 {
 	k_thread_create(&ecc_thread_data, ecc_thread_stack,
 			K_THREAD_STACK_SIZEOF(ecc_thread_stack), ecc_thread,
-			NULL, NULL, NULL, K_PRIO_PREEMPT(10), 0, K_NO_WAIT);
+			NULL, NULL, NULL, CONFIG_BT_ECC_THREAD_PRIO, 0,
```
Comment (jhedberg) [inline subsys/bluetooth/host/hci_ecc.c]: How come this is dropping use of the K_PRIO_PREEMPT() macro?

File: subsys/bluetooth/host/Kconfig [hunk 2]
```diff
@@ -512,6 +512,22 @@ config BT_TINYCRYPT_ECC
 	  to enabled for a combined build with Zephyr's own controller, since it
 	  does not have any special ECC support itself (at least not currently).
 
+if BT_TINYCRYPT_ECC
+config BT_ECC_THREAD_STACK_SIZE
+	int "Size of the ecc thread stack"
```
Comment (jhedberg) [inline subsys/bluetooth/host/Kconfig]: ECC, since it's an acronym, should be capitalized

File: subsys/bluetooth/host/Kconfig [hunk 3]
```diff
@@ -512,6 +512,22 @@ config BT_TINYCRYPT_ECC
...
+	default 1100
+	range 512 65536
+	help
+	  Size of the ecc thread stack.
```
Comment (jhedberg) [inline subsys/bluetooth/host/Kconfig]: Same here.

File: subsys/bluetooth/host/Kconfig [hunk 4]
```diff
@@ -512,6 +512,22 @@ config BT_TINYCRYPT_ECC
...
+
+config BT_ECC_THREAD_PRIO
+	int "Priority of ecc thread"
```
Comment (jhedberg) [inline subsys/bluetooth/host/Kconfig]: And here.

File: subsys/bluetooth/host/Kconfig [hunk 5]
```diff
@@ -512,6 +512,22 @@ config BT_TINYCRYPT_ECC
...
+	default 10
+	help
+	  Priority at which the ecc thread runs.
```
Comment (jhedberg) [inline subsys/bluetooth/host/Kconfig]: And here.

**Review by jhedberg — CHANGES_REQUESTED** [review]
Why is the commit message prefixed with "samples:"? It doesn't seem to touch any samples. Also, the commit message fails to answer the most important question that a commit message is supposed to answer, i.e. "why?". So far we have a single ECC library whose stack consumption the app can't influence. That's why this hasn't been configurable so far.

**Comment by jhedberg** [comment]
@pdunaj any update on this one?

**Comment by pdunaj** [comment]
Hi @jhedberg , I will fix it and push the change on Monday. It would be good to close it finally.

**Comment by aescolar** [comment]
@pdunaj any update?

**Comment by pdunaj** [comment]
Hi @aescolar , thanks for pinging me. I will try to rebase it and fix asap (today if possible).

**Comment by aescolar** [comment]
@pdunaj ping

**Comment by aescolar** [comment]
Some of what this PR wanted to do, seems to have been done in fddd786dfe9ab83f8c439bdc3142a0c1239b1fe9
#18324.
As there has been no activity in this PR for quite some time now I close it under the assumption that the author does not want to pursue it further. @pdunaj  If you do want to take it further please reopen it.

**Pull request closed without merging**